core/reactor_common.c

```diff
@@ -1439,15 +1439,20 @@ lf_token_t* _lf_set_new_array_impl(lf_token_t* token, size_t length, int num_des
 /**
  * Check the deadline of the currently executing reaction against the
  * current physical time. If the deadline has passed, invoke the deadline
- * handler and return true. Otherwise, return false.
- * 
+ * handler (if invoke_deadline_handler parameter is set true) and return true.
+ * Otherwise, return false.
+ *
  * @param self The self struct of the reactor.
+ * @param invoke_deadline_handler When this is set true, also invoke deadline
+ *  handler if the deadline has passed.
  * @return True if the specified deadline has passed and false otherwise.
  */
-bool _lf_check_deadline(self_base_t* self) {
+bool _lf_check_deadline(self_base_t* self, bool invoke_deadline_handler) {
 	reaction_t* reaction = self->executing_reaction;
 	if (get_physical_time() > get_logical_time() + reaction->deadline) {
-		reaction->deadline_violation_handler(self);
+	    if (invoke_deadline_handler) {
+		    reaction->deadline_violation_handler(self);
```

Comment:
Indentation isn't right. Tabs?

Reply:
Thanks for catching it! Fixed this in https://github.com/lf-lang/reactor-c/pull/46/commits/0682f42656f1b48bfe104a2b584dd244895c3857